garrysmod/lua/includes/extensions/vector.lua

```diff
@@ -3,10 +3,262 @@ local meta = FindMetaTable( "Vector" )
+	v:Add(...)
+	return v
+end
+
+--[[---------------------------------------------------------
+Adds a unpacked x, y and z to a vector
+-----------------------------------------------------------]]
+function meta:AddUnpacked(x, y, z)
+	self:AddUnpacked(self[1] + x, self[2] + y, self[3] + z)
+end
+
+--[[---------------------------------------------------------
+Adds a unpacked x, y and z to a copy vector
+-----------------------------------------------------------]]
+function meta:GetAddUnpacked(...)
+	local v = Vector(self)
+	v:AddUnpacked(...)
+	return v
+end
+
+--[[---------------------------------------------------------
+Returns a copy of a subracted vector
+-----------------------------------------------------------]]
+function meta:GetSub(...)
+	local v = Vector(self)
+	v:Sub(...)
+	return v
+end
+
+--[[---------------------------------------------------------
+Subtracts a unpacked x, y and z from a vector
+-----------------------------------------------------------]]
+function meta:SubUnpacked(x, y, z)
+	self:SubUnpacked(self[1] - x, self[2] - y, self[3] - z)
+end
+
+--[[---------------------------------------------------------
+Adds a unpacked x, y and z to a copy vector
+-----------------------------------------------------------]]
+function meta:GetSubUnpacked(...)
+	local v = Vector(self)
+	v:SubUnpacked(...)
+	return v
+end
+
+--[[---------------------------------------------------------
+Returns a copy of a divided vector
+-----------------------------------------------------------]]
+function meta:GetDiv(...)
+	local v = Vector(self)
+	v:Div(...)
+	return v
+end
+
+--[[---------------------------------------------------------
+Converts the vector to bisector of two vectors
+Converts vector to result of a*|b| + b*|a|
+-----------------------------------------------------------]]
+function meta:Bisect(vec)
+	local na, nb = self:Length(), vec:Length()
+	self:Mul(nb)
+	self:Add(vec:GetMul(na))
+end
+
+--[[---------------------------------------------------------
+Returns a copy of bisector of two vectors
+-----------------------------------------------------------]]
+function meta:GetBisected(...)
+	local v = Vector(self)
+	v:Bisect(...)
+	return v
+end
 
-	return Color( self.x * 255, self.y * 255, self.z * 255 )
+--[[---------------------------------------------------------
+Convets the vector to nagated version of itself
+-----------------------------------------------------------]]
+function meta:Negate()
+	local x, y, z = self:Unpack()
+	self:SetUnpacked(-x, -y, -z)
+end
+
+--[[---------------------------------------------------------
+Returns a copy of the nagated vector
+-----------------------------------------------------------]]
+function meta:GetNegated(...)
+	local v = Vector(self)
+	v:Negate(...)
+	return v
+end
+
+--[[---------------------------------------------------------
+Convets the vector to nagated version of itself
+-----------------------------------------------------------]]
+function meta:Nudge(num, vec)

```

Comment:
What are the use cases for this function? Where does the term "nudge" from?

I think it would just be better to write this inline as the multiplied vector will probably be reused in most cases.

Reply:
A short term for `origin = origin + coefficient * direction`

Reply:
Ah, so this is the same as [VectorMA](https://github.com/VSES/SourceEngine2007/blob/43a5c90a5ada1e69ca044595383be67f40b33c61/src_main/public/mathlib/mathlib.h#L269). Like Negate, this should be binded from the engine then instead of Lua.

Reply:
Ah, never new it has to be a dedicated engine function. I use it mery much in my addons, So I thought it must be needed here ;) Thank you !

Reply:
I will Remove Nudge. This is the exact thing I am talking about.